internal/api/health.go

```diff
@@ -0,0 +1,69 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/pass-wall/passwall-server/internal/config"
+	"github.com/pass-wall/passwall-server/internal/storage"
+)
+
+var (
+	// should be improved
+	Port          = config.SetupConfigDefaults().Server.Port
+	ServerAddress = "0.0.0.0" + ":" + Port
+)
+
+type HealthProp struct {
+	StatusCode int
+	Err        error
+}
+
+type Services struct {
+	API      *HealthProp
+	Database *HealthProp
+}
+
+func HealthCheck(s storage.Store) http.HandlerFunc {
+
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		var checkResult Services
+		var APIStatus *HealthProp
+		var DBStatus *HealthProp
+
+		if err := checkEndPoint(ServerAddress); err != nil {
```

Comment:
Do we really need to test already running server and endpoint? I think if this endpoint responds, then it means 0.0.0.0:port is up and running. Also this is not an external service or endpoint. Am I missing something?

Reply:
In my opinion, test case should include several states, like down, up  and error states. It would be much better to have test table to examine different states, right.

I think, I misunderstood the issue then, what do you mean exactly by external service or endpoint  ? @yakuter  I thought that making GET request without auth could be enough to check health status.

Reply:
Well you thought about testing all endpoints in the future with table design. It is OK for me. Thank you.